src/NuGet.Indexing/DownloadsByVersion.cs

```diff
@@ -48,8 +41,13 @@ public int Total
 
             set
             {
+                if (_downloadsByVersion.ContainsKey(version))
+                {
+                    _total = _total - _downloadsByVersion[version];
```

Comment:
Being out of sync while the data is being reloaded is still a major concern--we always want it to be in sync, and having intermittent sync issues is a nightmare to deal with as you experienced while debugging this issue.

You can still construct the new `DownloadsByVersion` using the old `DownloadsByVersion`. Clone the object, perform the modifications, and then swap. This would get rid of your concern and guarantee that the data is never out of sync.